src/nanoemoji/maximum_color.py

```diff
@@ -0,0 +1,179 @@
+        raise ValueError("Must have one of COLR, SVG")
+
+    if has_svg:
+        return "SVG "
+    if has_colr:
+        return "COLR"
+
+    raise ValueError("Impossible")
+
+
+def svg_extract_dir() -> Path:
+    return build_dir() / "svg_dump"
+
+
+def picosvg_dir() -> Path:
+    return build_dir() / "picosvg"
+
+
+def picosvg_dest(input_svg: Path) -> Path:
+    return picosvg_dir() / input_svg.name
+
+
+def _write_preamble(nw: NinjaWriter, input_font: Path):
+    module_rule(
+        nw,
+        "extract_svgs_from_otsvg",
+        f"--output_dir {rel_build(svg_extract_dir())} $in $out",
+    )
+    nw.newline()
+
+    module_rule(nw, "write_glyphmap_for_glyph_svgs", "$in > $out")
+    nw.newline()
+
+    module_rule(nw, "write_config_for_glyph_svgs", "$in $out")
+    nw.newline()
+
+    module_rule(
+        nw,
+        "write_font",
+        f"--glyphmap_file {_SVG2COLR_GLYPHMAP} --config_file {_SVG2COLR_CONFIG} --output_file $out",
+        rule_name="write_colr_font_from_svg_dump",
+    )
+    nw.newline()
+
+    nw.rule(
+        f"picosvg",
+        f"picosvg  --output_file $out $in",
+    )
+    nw.newline()
+
+    module_rule(
+        nw,
+        "glue_together",
+        f"--color_table COLR --target_font {rel_build(input_font)} --donor_font $in --output_file $out",
+        rule_name="copy_colr_from_svg2colr",
+    )
+    nw.newline()
+
+
+def _write_svg_extract(nw: NinjaWriter, input_font: Path, font: ttLib.TTFont):
+    # extract the svgs
+    svg_extracts = [
+        rel_build(svg_extract_dir() / f"{gid}.svg") for gid, _ in svg_glyphs(font)
+    ]
+    nw.build(svg_extracts, "extract_svgs_from_otsvg", input_font)
+    nw.newline()
+
+    # picosvg them
+    picosvgs = [rel_build(picosvg_dest(s)) for s in svg_extracts]
+    for svg_extract, picosvg in zip(svg_extracts, picosvgs):
+        nw.build(picosvg, "picosvg", svg_extract)
+    nw.newline()
+
+    # make a glyphmap
+    nw.build(
+        _SVG2COLR_GLYPHMAP, "write_glyphmap_for_glyph_svgs", picosvgs + [input_font]
+    )
+    nw.newline()
+
+    # make a config
+    nw.build(_SVG2COLR_CONFIG, "write_config_for_glyph_svgs", input_font)
+    nw.newline()
+
+    # generate a new font with COLR glyphs that use the same names as the original
+    nw.build(
+        "colr_from_svg.ttf",
+        "write_colr_font_from_svg_dump",
+        [_SVG2COLR_GLYPHMAP, _SVG2COLR_CONFIG],
+    )
+    nw.newline()
+
+    # stick our shiny new COLR table onto the input font and declare victory
+    nw.build(
+        input_font.name,
+        "copy_colr_from_svg2colr",
+        "colr_from_svg.ttf",
+    )
+    nw.newline()
+
+
+def main(argv):
+    if len(argv) != 2:
+        raise ValueError("Must have one argument, a font file")
+
+    input_font = Path(argv[1])
+    assert input_font.is_file()
+    font = ttLib.TTFont(input_font)
+
+    build_file = build_dir() / "build.ninja"
+    build_dir().mkdir(parents=True, exist_ok=True)
+
+    print(build_dir())  # TEMPORARY
```

Comment:
wondering if we need a pre-commit hook that fails if debug print statements are found

Reply:
a unit test that forbids them in unexpected locations? :D